frontend/src/metabase/query_builder/actions.js

```diff
@@ -1216,24 +1213,12 @@ export const loadObjectDetailFKReferences = createThunkAction(
       const queryResult = getFirstQueryResult(getState());
       const tableForeignKeys = getTableForeignKeys(getState());
 
-      function getObjectDetailIdValue(data) {
-        for (let i = 0; i < data.cols.length; i++) {
-          const coldef = data.cols[i];
-          if (isPK(coldef.semantic_type)) {
-            return data.rows[0][i];
-          }
-        }
-      }
-
       async function getFKCount(card, queryResult, fk) {
         const fkQuery = Q_DEPRECATED.createQuery("query");
         fkQuery.database = card.dataset_query.database;
         fkQuery.query["source-table"] = fk.origin.table_id;
         fkQuery.query.aggregation = ["count"];
-        fkQuery.query.filter = [
-          "and",
-          ["=", fk.origin.id, getObjectDetailIdValue(queryResult.data)],
-        ];
```

Comment:
@camsaul this `"and"` was doing nothing, right? I extracted this to a function since it was duplicated but both uses had the `"and"`.